mlflow/store/tracking/file_store.py

```diff
@@ -333,7 +333,7 @@ def search_experiments(
     def get_experiment_by_name(self, experiment_name):
```

Comment:
It seems `search_experiment` has parameter `view_type`. Shall we do the same not to change the default behavior?

https://github.com/mlflow/mlflow/blame/874c23c547c12b5e5743a014fa18c5557d7d9b11/mlflow/store/tracking/file_store.py#L287 

One regression case seems to be `set_experiemnt`. We use `get_experiment_by_name` to check if we need to create a new one or not. Currently it creates a new one when the experiment with that name is not active.
https://github.com/mlflow/mlflow/blob/36227fcfcee9656c807bbb4180e5829ff04faf66/mlflow/tracking/fluent.py#L143-L153

Reply:
Actually it's matching the behavior with sql backend store. set_experiment shouldn't succeed if a deleted experiment with the same name exists. It should reach this check https://github.com/mlflow/mlflow/blob/36227fcfcee9656c807bbb4180e5829ff04faf66/mlflow/tracking/fluent.py#L162 and raise error. I think it's a bug fix instead of regression, WDYT?

Reply:
Ahh ok so only file store filters to `ACTIVE_ONLY` now. So we don't allow creating a new experiment with the same name as deleted one, across all stores.